### ISSUE TITLE
Print final Query sent to Elasticsearch

### ISSUE THREAD

**Issue by vmogilev** [elastic/elasticsearch-dsl-py#448]
Is there anyway to print the final Query sent to Elasticsearch?

Thank you.

**Comment by HonzaKral** [comment]
You can always ask the `Search` object for it's raw representation by calling the `.to_dict()` method. If you want to log what's going on I'd recommend that you enable logging for the `elasticsearch-py` library (0) by using the standard logging module. You can either choose the `elasticsearch` logger which logs exactly what's going on (set the log level to `INFO` to see all requests) or `elasticsearch.trace` for a more friendly representation where all calls to elasticsearch will be logged as `curl` commands against `localhost:9200` with pretty printed json.

0 - http://elasticsearch-py.readthedocs.io/en/master/#logging

**Comment by vmogilev** [comment]
Thank you, that's very helpful.

While waiting for your answer, I hacked the following which helped to understand how the Facetted DSL query is formed:

    pp = pprint.PrettyPrinter(indent=4)
    logger.debug(pp.pformat(s.build_search().to_dict()))

**Comment by gitanupam** [comment]
For reference, this [StackOverflow post](https://stackoverflow.com/a/33410967/1526703) shows how you can turn on logs dynamically.

```
PUT /_settings
{
  "index": {
    "search.slowlog.level": "trace",
    "search.slowlog.threshold.query.trace": "1ms"
  }
}


```

**Comment by ha7ilm** [comment]
TL;DR you just have to set up Python logging before running the query and you will find the full query in JSON there.

```python3
import logging
logging.basicConfig(filename='myfile.log', level=logging.DEBUG)
# ...
res = es.search(index="my_index", body={ #...
```

**Comment by Condescending-dompurify3** [comment]
Here’s what https://lnkr.fun/da4b51d I was looking for
